tools/expotools/src/commands/AndroidBuildPackages.ts

```diff
@@ -0,0 +1,345 @@
+  }
+}
+
+async function _updateExpoViewAsync(packages: Package[], sdkVersion: string): Promise<void> {
+  let appBuildGradle = path.join(ANDROID_DIR, 'app', 'build.gradle');
+  let expoViewBuildGradle = path.join(ANDROID_DIR, 'expoview', 'build.gradle');
+  const settingsGradle = path.join(ANDROID_DIR, 'settings.gradle');
+  const constantsJava = path.join(
+    ANDROID_DIR,
+    'expoview/src/main/java/host/exp/exponent/Constants.java'
+  );
+  const multipleVersionReactNativeActivity = path.join(
+    ANDROID_DIR,
+    'expoview/src/main/java/host/exp/exponent/experience/MultipleVersionReactNativeActivity.java'
+  );
+
+  // Modify permanently
+  await _regexFileAsync(expoViewBuildGradle, /version = '[\d.]+'/, `version = '${sdkVersion}'`);
+  await _regexFileAsync(
+    expoViewBuildGradle,
+    /api 'com.facebook.react:react-native:[\d.]+'/,
+    `api 'com.facebook.react:react-native:${sdkVersion}'`
+  );
+  await _regexFileAsync(
+    path.join(ANDROID_DIR, 'ReactAndroid', 'release.gradle'),
+    /version = '[\d.]+'/,
+    `version = '${sdkVersion}'`
+  );
+  await _regexFileAsync(
+    path.join(ANDROID_DIR, 'app', 'build.gradle'),
+    /host.exp.exponent:expoview:[\d.]+/,
+    `host.exp.exponent:expoview:${sdkVersion}`
+  );
+
+  const filesToStash = [
+    appBuildGradle,
+    expoViewBuildGradle,
+    multipleVersionReactNativeActivity,
+    constantsJava,
+    settingsGradle,
+  ];
+  await _stashFilesAsync(filesToStash);
+
+  // Modify temporarily
+  await _regexFileAsync(
+    constantsJava,
+    /TEMPORARY_ABI_VERSION\s*=\s*null/,
+    `TEMPORARY_ABI_VERSION = "${sdkVersion}"`
+  );
+  await _regexFileAsync(
+    settingsGradle,
+    `// FLAG_BEGIN_REMOVE__UPDATE_EXPOKIT`,
+    `/*`
+  );
+  await _regexFileAsync(
+    settingsGradle,
+    `// FLAG_END_REMOVE__UPDATE_EXPOKIT`,
+    `*/ //`
+  );
+  await _uncommentWhenDistributing([appBuildGradle, expoViewBuildGradle]);
+  await _commentWhenDistributing([constantsJava, expoViewBuildGradle, multipleVersionReactNativeActivity]);
+
+  // Clear maven local so that we don't end up with multiple versions
+  console.log(' ❌  Clearing old package versions...')
+
+  await spawnAsync('rm', [
+    '-rf',
+    ...packages.map(pkg => path.join(process.env.HOME!, '.m2', 'repository', pkg.buildDirRelative)),
+    ...packages.map(pkg => path.join(ANDROID_DIR, 'maven', pkg.buildDirRelative)),
+    ...packages.map(pkg => path.join(pkg.sourceDir, 'build'))
+  ]);
+
+  for (const pkg of packages) {
+    process.stdout.write(` 🛠   Building ${pkg.name}...`);
+    await spawnAsync('./gradlew', [`:${pkg.name}:uploadArchives`], {
+      cwd: ANDROID_DIR,
+    });
+    readline.clearLine(process.stdout, 0);
+    readline.cursorTo(process.stdout, 0);
+    process.stdout.write(` ✅  Finished building ${pkg.name}\n`);
+  }
+
+  await _restoreFilesAsync(filesToStash);
+
+  console.log(' 🚚  Copying newly built packages...');
+
+  await spawnAsync('mkdir', ['-p', path.join(ANDROID_DIR, 'maven/com/facebook')])
```

Comment:
`fs-extra` has a method that does the same as `mkdir -p` which would look better than `spawnAsync` and works on all platforms:
```suggestion
  await fs.mkdirp(path.join(ANDROID_DIR, 'maven/com/facebook'));
```

Reply:
Node 10 supports the `recursive` option: https://nodejs.org/docs/latest-v10.x/api/fs.html#fs_fs_mkdir_path_options_callback. IIRC we use Node 10.16 in the Expo repo and can rely on this option.